growthbook/growthbook
# Add support for big saved groups to JS sdk

### Features and Changes

Splits the JS sdk logic out of #2616 so that it can be merged first. The new logic is all backwards compatible, so this will allow us to release the new 1.0.2 version of the sdk ahead of time and prevent any issues with the proxy server.


### Testing
Setup:
1. Checkout the `kc/saved-groups-improvements` branch to grab the code that generates the new style of payload
2. Run `yarn link` in the `sdk-js` package and start running `yarn dev`
3. Clone the https://github.com/growthbook/kevin-sdk-testing/tree/main/sdk-js repo and run `yarn dev` there

Changing the saved group & feature definitions in your app should result in the correct feature evaluation in the next app.
